src/adapters/rubic/execution/src/index.ts

```diff
@@ -19,30 +19,32 @@ export const getUserTransactionData = async (startBlock: number, endBlock: numbe
   console.log(`Get Tx Data From ${startBlock} to ${endBlock}`)
 
   const manager = new RubicQueriesManager(startBlock, endBlock)
-  const [onchainSwaps, ccrSwaps] = await Promise.all([
-      manager.queryRubicSwappedGenerics(),
-      manager.queryRubicTransferStarteds()
+  const results = await Promise.all([
+    manager.queryOnChainInfo(),
+    manager.queryCrossChainInfo()
   ])
-  const transactions = [...onchainSwaps, ...ccrSwaps]
+  const transactions = results.flat()
 
-  return transactions
+  return transactions;
 };
 
 class RubicQueriesManager {
-  private _provider = new JsonRpcProvider('https://rpc.zklink.io')
+  private _provider = new ethers.providers.JsonRpcProvider('https://rpc.zklink.io')
 
```

Comment:
TS error, in etherjs v6, `JsonRpcProvider` should be imported from ethers directly.